### ISSUE TITLE
Error in TYPO3 11.5.30

### ISSUE THREAD

**Issue by Scopestyle** [pixelschrubber/toc#8]
Hi! I'm trying to install toc 0.9.0 (TER Version) in my TYPO3 11.5.30 Installation. The following error occurs in Frontend:

```
array_combine(): Argument #1 ($keys) and argument #2 ($values) must have the same number of elements
```

The log notes the following information:

```
[CRITICAL] request="cf1f43f405b49" component="TYPO3.CMS.Core.Error.ProductionExceptionHandler": Core: Exception handler (WEB: FE): ValueError, code #0, file /html/typo3/typo3conf/ext/toc/Classes/Domain/Repository/TOCRepository.php, line 118: array_combine(): Argument #1 ($keys) and argument #2 ($values) must have the same number of elements - {"mode":"WEB","application_mode":"FE","exception_class":"ValueError","exception_code":0,"file":"/html/typo3/typo3conf/ext/toc/Classes/Domain/Repository/TOCRepository.php","line":118,"message":"array_combine(): Argument #1 ($keys) and argument #2 ($values) must have the same number of elements","request_url":"https://www.*redacted*.com/test/","exception":null}
```

Is this an error in the extension, or something I'm missing on my end?

**Comment by pixelschrubber** [comment]
Sorry, currently not much time to test it myself. Did you try to change the settings in order to use it without hierarchy? You can toggle the setting to 0 for the setting 'Enable-Hierarchy' - afterwards this code part won't be executed. 

Other than that a rather simple question: Do you have headlines on the page?

**Comment by Scopestyle** [comment]
Yeah there are headlines on the page, just not enough 😄 Thank you for pointing me in the right direction. The culprit was ```settings.Show-Amount``` which has a default value of 4. I had less on the page I tried to get the extension to work. Setting this to 1 fixed it.

I did encounter an error in how the Javascript executes on pages where the extension is not active. Here the extension broke all other JS on the page. My workaround for this was wrapping the extension's JS code in a *DOMContentLoaded* function, so the code only executes after the DOM is fully loaded. 

```toc/Resources/Public/Script/toc-script.js```

```
document.addEventListener('DOMContentLoaded', function() {
   // New Toggle Style
   ...
   // Jump to top
   ...
   // Speaking URL hashes
   ...
   // Jumplinks
   ...
   // Browserhistory
   ...
});
```

I'll play around with it some more. Thanx again!
